### ISSUE TITLE
Allow patched ROMs to be run in Einstein

### ISSUE THREAD

**Issue by FrankFromPdaSoft** [pguyot/Einstein#166]
Currently Einstein only loads ROM files whose CRC value matches expected values. Since a ROM patch changes the CRC value, it can't be loaded. It would be cool, though, if this was possible so that patched ROM files can be tested using Einstein.
To ensure that this option can only be used by people who know exactly what they are doing, the availability of this option should be controlled by something that isn't part of the default setup. This could, for example, be the presence of a particular file in the Einstein directory that the user must create.
This change would probably need to be in the ComputeROMId(KUInt8* inROMPtr) method of the TROMImage.cpp class.

**Comment by MatthiasWM** [comment]
It should load the ROM, even if the checksum is not recognised. It checks for the size of the ROM which must still be exactly 8MB. There is another issue though. If the checksum is not recognised, then a bunch of Einstein internal patches are not applied. They are not essential, but they do help Einstein to run smoother. In a patched US ROM, time and date are no longer set, for example.

**Comment by MatthiasWM** [comment]
Serach for `T_ROM_PATCH` if you are interested to see what's being patched.

**Comment by FrankFromPdaSoft** [comment]
You are right, this was a wrong assumption. Changed ROMs are loaded. What's not loaded is the file from my weird ROM board (https://www.pda-soft.de/weirdromboard.html) although it has the correct size. Einstein doesn't get past the boot sequence. The last I see is this:
![image](https://user-images.githubusercontent.com/10109953/162410942-06a742ce-5483-46ce-bb48-23cf574706b5.png)
Right clicks still work, the menu appears, so Einstein itself doesn't seem to be crashed.

**Comment by MatthiasWM** [comment]
Please send me a copy. I can check where it gets stuck. Maybe (but unlikely) it's something simple.

**Comment by chuma** [comment]
Did y'all ever get anywhere with this?  I've been doing quite a bit of analysis of the boot sequence of NOS 2.1 lately, I could help.

**Comment by MatthiasWM** [comment]
No, we did not follow up on this. Maybe you can ask Frank directly if he wants to share a copy of that early ROM. I don't have the file (or at least I can't find it). As far as I can remember, it was a pre 2.0 with no grayscale. But that's as far as I got.
